lua/refactoring/refactor/106.lua

```diff
@@ -64,44 +64,47 @@ local function get_extract_setup_pipeline(bufnr)
         :add_task(get_selected_local_defs)
 end
 
+local function extract_setup(refactor)
+    local selected_local_references = get_selected_local_references(refactor)
+
+    local function_name = get_input("106: Extract Function Name > ")
+    assert(function_name ~= "", "Error: Must provide function name")
+
+    local function_body = refactor.region:get_text(refactor.region.bufnr)
+    table.insert(function_body, refactor.code["return"]("fill_me"))
+    local args = vim.fn.sort(vim.tbl_keys(selected_local_references))
+
+    local function_code = refactor.code["function"]({
+        name = function_name,
+        args = args,
+        body = function_body,
+    })
+
+    refactor.text_edits = {
+        {
+            region = utils.region_above_node(refactor.scope),
+            text = function_code,
+            bufnr = refactor.buffers[2],
```

Comment:
I assume this works only because bufnr[2] doesn't exist on same file operations?   Is that correct?

Reply:
I am not the biggest fan of this approach.  Is there a better way?